lib/rubocop/cop/layout/trailing_blank_lines.rb

```diff
@@ -5,6 +5,44 @@ module Cop
     module Layout
       # This cop looks for trailing blank lines and a final newline in the
       # source code.
+      #
+      # @example EnforcedStyle: final_blank_line
+      #   # `final_blank_line` looks for one blank line followed by a new line
+      #   # at the end of files.
+      #
+      #   # bad
+      #   class Foo; end
+      #
+      #   __END__
+      #
+      #   # bad
+      #   class Foo; end
+      #   __END__
+      #
+      #   # good
+      #   class Foo; end
+      #   # a blank line
+      #
+      #   __END__
+      #
+      # @example EnforcedStyle: final_newline (default)
+      #   # `final_newline` looks for one newline at the end of files.
+      #
+      #   # bad
+      #   class Foo; end
+      #   # a blank line
+      #
+      #   __END__
+      #
+      #   # bad
+      #   class Foo; end
+      #   __END__
+      #
+      #   # good
+      #   class Foo; end
+      #
+      #   __END__
```

Comment:
Since `__END__` is a meaningful word in Ruby, this description may be misleading to users.
https://ruby-doc.org/core-2.5.0/Object.html#DATA

The following is an idea to use `# EOF` instead of` __END__`.

```diff
 # @example EnforcedStyle: final_newline (default)
 #   # `final_newline` looks for one newline at the end of files.
 #
 #   # bad
-#   class Foo; end
-#   # a blank line
-#
-#   __END__
+#   class Foo; end
+#
+#   # EOF
 #
 #   # bad
-#   class Foo; end
-#   __END__
+#   class Foo; end # EOF
 #
 #   # good
-#   class Foo; end
-#
-#   __END__
+#   class Foo; end
+#   # EOF
 #
```

Since the example code aims to be executable, It use source code comment. However, I'm not so sure if this is easy to understand 😟

Reply:
Thanks!  You're absolutely right.  I wasn't familiar with `__END__`, but saw that the test file uses it.  [Based on this article](https://til.hashrocket.com/posts/17787bf181-rubys-end), it makes sense that this works for the tests and in practice. But doesn't actually mean the end of a file. (And now I know something new 🎉 !)

# EOF is definitely an improvement.